src/core_plugins/tagcloud/public/tag_cloud_visualization.js

```diff
@@ -0,0 +1,132 @@
+import TagCloud from 'plugins/tagcloud/tag_cloud';
+import AggConfigResult from 'ui/vis/agg_config_result';
+import { Observable } from 'rxjs';
+
+const MAX_TAG_COUNT = 200;
+
+export class TagCloudVisualization {
+
+  constructor(node, vis) {
+    this._containerNode = node;
+
+    const nodeContens = document.createElement('div');
+    nodeContens.innerHTML = `
+            <div class="tagcloud-vis">
+              <div class="tagcloud-custom-label"></div>
+              <div class="tagcloud-notifications">
+                  <div class="tagcloud-truncated-message">
+                    The number of tags has been truncated to avoid long draw times.
+                  </div>
+                  <div class="tagcloud-incomplete-message">
+                    The container is too small to display the entire cloud. Tags might be cropped or omitted.
+                  </div>
+              </div>
+            </div>
+          `;
+
+    this._containerNode.appendChild(nodeContens);
+
+    this._vis = vis;
+    this._bucketAgg = null;
+    this._truncated = false;
+    this._tagCloud = new TagCloud(node);
+    this._tagCloud.on('select', (event) => {
+      if (!this._bucketAgg) {
+        return;
+      }
+      const aggConfigResult = new AggConfigResult(this._bucketAgg, false, event, event);
+      this._vis.API.events.filter({ point: { aggConfigResult: aggConfigResult } });
+    });
+    this._renderComplete$ = Observable.fromEvent(this._tagCloud, 'renderComplete');
+
+  }
+
+  async render(data, status) {
+
+
+    if (status.params || status.aggs) {
+      this._updateParams();
+    }
+
+    if (status.data) {
+      this._updateData(data);
+    }
+
+
+    if (status.resize) {
+      this._resize();
+    }
+
+    await this._renderComplete$.take(1).toPromise();
```

Comment:
Observables don't start "listening" until they are subscribed to, which is being done by the `.toPromise` call. If the `renderComplete` event was fired before we got to this line, then this isn't going to do what we want.

Reply:
For each render() call, we're only interested in the first render-complete-call starting after line 59. Do we need to subscribe to that observable up front, or at the start of the method, or in the constructor when we create it?

Reply:
If the `renderComplete` is only called after line 59 then the current approach should work.